packages/admin-ui/src/calendar/calendar-state.ts

```diff
@@ -0,0 +1,320 @@
+    setFocusedDate,
+    focusNextDay() {
+      focusCell(addDays(focusedDate, 1))
+    },
+    focusPreviousDay() {
+      focusCell(subDays(focusedDate, 1))
+    },
+    focusNextWeek() {
+      focusCell(addWeeks(focusedDate, 1))
+    },
+    focusPreviousWeek() {
+      focusCell(subWeeks(focusedDate, 1))
+    },
+    focusNextMonth() {
+      focusCell(addMonths(focusedDate, 1))
+    },
+    focusPreviousMonth() {
+      focusCell(subMonths(focusedDate, 1))
+    },
+    focusStartOfMonth() {
+      focusCell(startOfMonth(focusedDate))
+    },
+    focusEndOfMonth() {
+      focusCell(endOfMonth(startOfDay(focusedDate)))
+    },
+    focusNextYear() {
+      focusCell(addYears(focusedDate, 1))
+    },
+    focusPreviousYear() {
+      focusCell(subYears(focusedDate, 1))
+    },
+    selectFocusedDate() {
+      setDate(focusedDate)
+    },
+    selectDate(date: Date) {
+      setDate(date)
+    },
+    isInvalidDateRange,
+  }
+}
+
+export interface CalendarState {
+  /**
+   * Id for the Calendar Header
+   */
+  calendarId: string | undefined
+  /**
+   * Selected Date value
+   */
+  dateValue: Date
+  /**
+   * Month of the current date value
+   */
+  month: number
+  /**
+   * Year of the current date value
+   */
+  year: number
+  /**
+   * Start of the week for the current date value
+   */
+  weekStart: number
+  /**
+   * Generated week days for CalendarWeekTitle based on weekStart
+   */
+  weekDays: Array<{
+    title: string
+    abbr: string
+  }>
+  /**
+   * Generated days in the current month
+   */
+  daysInMonth: Date[][]
+  /**
+   * `true` if the calendar is disabled
+   */
+  isDisabled: boolean
+  /**
+   * `true` if the calendar is focused
+   */
+  isFocused: boolean
+  /**
+   * `true` if the calendar is only readonly
+   */
+  isReadOnly: boolean
+  /**
+   * Month of the current Date
+   */
+  currentMonth: Date
+  /**
+   * Date value that is currently focused
+   */
+  focusedDate: Date
+  /**
+   * Informs if the given date is within the min & max date.
+   */
+  isInvalidDateRange: (value: Date) => boolean
+}
+
+export interface CalendarActions {
+  /**
+   * Sets `isFocused`
+   */
+  setFocused: Dispatch<SetStateAction<boolean>>
+  /**
+   * Sets `currentMonth`
+   */
+  setCurrentMonth: Dispatch<SetStateAction<Date>>
+  /**
+   *  Sets `focusedDate`
+   */
+  setFocusedDate: Dispatch<SetStateAction<Date>>
+  /**
+   * Sets `dateValue`
+   */
+  setDateValue: (value: Date) => void
+  /**
+   * Focus the cell of the specified date
+   */
+  focusCell: (value: Date) => void
+  /**
+   * Focus the cell next to the current date
+   */
+  focusNextDay: () => void
+  /**
+   * Focus the cell prev to the current date
+   */
+  focusPreviousDay: () => void
+  /**
+   * Focus the cell one week next to the current date
+   */
+  focusNextWeek: () => void
+  /**
+   * Focus the cell one week prev to the current date
+   */
+  focusPreviousWeek: () => void
+  /**
+   * Focus the cell one month next to the current date
```

Comment:
Not clear to me if this focuses current day + 1 month, or if it focuses on the 1st of the next month. Same for year/week

Reply:
Is the current day + 1 month